### PR TITLE
databases-week1

### PR DESCRIPTION
**Pull request by Mamidi001**
Done with database week1 home work

### CODE REVIEW

**Review by MLMeyer — APPROVED** [review]
Great Work! 🙌💪🏼🔥

For a better structure and easier overview, I would suggest you place the comment, with numeration, above each query 😊

In assignment 4 “find all the tasks that are not marked as done” 
It is more custom to relocate the “NOT” in this order: 
- WHERE status.name NOT LIKE “Done”1;

Since you know, that the status.name is in fact “Done” you could also write 
WHERE status.name != “Done”;  
You can play around with it if you want 😊

In the last assignment, assignment 10, you want to show the status name with the most assignments first. Do you want to your ASC or DESC? Try both and see what you get 💡😊

File: databases/week1/hyf_lesson1.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
+SELECT * -- get the table
+FROM task;
+SELECT COUNT(title) --how many tasks are in the task table
```
Comment (martamatos) [inline databases/week1/hyf_lesson1.sql:3]: What if some tasks have a title that is NULL?

File: databases/week1/hyf_lesson1.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+SELECT COUNT(id) -- count of do not have valid date
+FROM task
+WHERE due_date is NULL;
+SELECT task.title, --all the tasks that are marked as done
+status.name
+FROM task JOIN status on task.status_id = status.id 
+WHERE status.name LIKE'Done';
+SELECT task.title, -- all the tasks that are not marked as done
+status.name
+FROM task JOIN status on task.status_id <> status.id
```
Comment (martamatos) [inline databases/week1/hyf_lesson1.sql:14]: Are you sure about the `<>`? What does it mean?

File: databases/week1/hyf_lesson1.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+FROM
+  task
+  JOIN status ON task.status_id = status.id;
+SELECT status.name, COUNT(task.id)--Get the name of each status, along with a count of how many tasks have that status
+FROM status
+JOIN task ON status.id = task.status_id
+GROUP BY status.name;
+SELECT status.name, COUNT(task.id) --Get the names of all statuses, sorted by the status with most tasks first
+FROM task
+INNER JOIN status ON status_id= task.status_id
```
Comment (martamatos) [inline databases/week1/hyf_lesson1.sql:38]: I think you meant status.id instead of status_id?

File: databases/week1/hyf_lesson1.sql (hunk 2 above)
Comment (Mamidi001) [reply databases/week1/hyf_lesson1.sql:14]: <> it means not equal to (!=)

File: databases/week1/hyf_lesson1.sql (hunk 3 above)
Comment (Mamidi001) [reply databases/week1/hyf_lesson1.sql:38]: I got the out correct. if i use status.id it throws an error.

**Comment by Mamidi001** [comment]
modified after mentor feedback

File: databases/week1/hyf_lesson1.sql (hunk 2 above)
Comment (martamatos) [reply databases/week1/hyf_lesson1.sql:25]: Exactly, and is that what you want? 
What kind of results did you get from that query?

File: databases/week1/hyf_lesson1.sql (hunk 3 above)
Comment (martamatos) [reply databases/week1/hyf_lesson1.sql]: Which error did you get? o.O Can you share a printscreen?

File: databases/week1/hyf_lesson1.sql [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+FROM status
+JOIN task ON status.id = task.status_id
+GROUP BY status.name;
+
+--Get the names of all statuses, sorted by the status with most tasks first
+SELECT status.name, COUNT(task.id) 
+FROM task
+INNER JOIN status ON status.id= task.status_id
+GROUP BY status_id
+ORDER BY status_id, COUNT(task.id) DESC;
```
Comment (martamatos) [inline databases/week1/hyf_lesson1.sql:63]: And in the meantime i also noticed that this query is not returning the correct result, can you figure out why? (hint: you only need to use one attribute in the order by)

**Pull request merged**